src/utils/cosmosdb_model.ts

```diff
@@ -30,6 +29,30 @@ export const BaseModel = t.interface({
 
 export type BaseModel = t.TypeOf<typeof BaseModel>;
 
+/**
+ * Model a tuple which defines the references to search for a document.
+ * A Cosmodb document must be looked-up by its Identity alongside its PartitionKey. If PartitionKey field is the same of Identity field, it can be omitted.
+ * Hence this type models both cases: (ID) or (ID,PK) respectively if PK literal type is omitted or provided.
+ * @param T the type of the document mapped by the model
+ * @param ID the literal type defining the ID field for the document.
+ * @param PK (optional) the literal type defining the eventual partition key for the document. Default: undefined
+ */
+export type DocumentSearchKey<
+  // T might not include base model fields ("id"), but we know they are mandatory in cosmos documents
+  // Quick win would be to narrow T to extend BaseModel, but that way we'd lose usage flexibility
```

Comment:
what do you mean with " we'd lose usage flexibility" ?

Reply:
tl;dr you can pass `MyModel` and not necessarily `RetrievedMyModel`

`T` is the struct which define our model schema. It's not required to have a `id` field. Then we have some type definitions that extend (or _compose with_) `T` and they do have such field, but they don't overlap each other. So you would have needed to oveload `DocumentSearchKey` or any child type - a little mess.
I thought that the easier interface was to use the minimum subset of definition (that is: `T`) and then just make the constrant that `ID` and `PK` must fall in the set `keyof T | "id"`.

I hope that makes sense ;)